### PR TITLE
Fixes an error in parameters order that resulted in factions not being able to be set up properly

### PR DESCRIPTION
**Pull request by ShiftyRail**
You had : 
```dm
var/datum/role/newRole = new initroletype(null, src, initial_role)
```
With `src` being a faction, but the role's `New()` was : 
```dm
/datum/role/New(var/datum/mind/M, var/datum/faction/fac=null, var/new_id, var/override = FALSE)
```
IE, a `faction` was expected as the second parameter.
I also made a bunch of error runtime instead of just a warning to have an easier time tracking down the issue.

### CODE REVIEW

File: code/datums/gamemode/gamemode.dm [hunk 1]
```diff
@@ -93,7 +93,7 @@
 			if(!P.client.desires_role(F.required_pref) || jobban_isbanned(P, F.required_pref))
 				continue
 			if(!F.HandleNewMind(P.mind))
-				warning("[P.mind] failed [F] HandleNewMind!")
+				CRASH("[P.mind] failed [F] HandleNewMind!")
 				continue
```
Comment (DamianX) [inline code/datums/gamemode/gamemode.dm]: dead code, CRASH will stop the proc

File: code/datums/gamemode/role/role.dm [hunk 2]
```diff
@@ -122,10 +122,10 @@
 
 /datum/role/proc/AssignToRole(var/datum/mind/M,var/override = 0)
 	if(!istype(M) && !override)
-		WARNING("M is [M.type]!")
+		CRASH("M is [M.type]!")
 		return 0
```
Comment (DamianX) [inline code/datums/gamemode/role/role.dm]: dead code, CRASH will stop the proc

File: code/datums/gamemode/role/role.dm [hunk 3]
```diff
@@ -122,10 +122,10 @@
...
 	if(!CanBeAssigned(M) && !override)
-		WARNING("[M] was to be assigned to [name] but failed CanBeAssigned!")
+		CRASH("[M] was to be assigned to [name] but failed CanBeAssigned!")
 		return 0
```
Comment (DamianX) [inline code/datums/gamemode/role/role.dm]: dead code, CRASH will stop the proc

File: code/datums/gamemode/factions/faction.dm [hunk 4]
```diff
@@ -66,12 +66,12 @@ var/list/factions_with_hud_icons = list()
 /datum/faction/proc/HandleNewMind(var/datum/mind/M) //Used on faction creation
 	for(var/datum/role/R in members)
 		if(R.antag == M)
-			WARNING("Mind was already a role in this faction")
+			warning("Mind was already a role in this faction")
```
Comment (DamianX) [inline code/datums/gamemode/factions/faction.dm]: Why would you change these from `WARNING` (which shows file, line number, src, usr) to `warning` (which only shows the message)?

File: code/datums/gamemode/factions/faction.dm [hunk 5]
```diff
@@ -81,7 +81,7 @@ var/list/factions_with_hud_icons = list()
 /datum/faction/proc/HandleRecruitedMind(var/datum/mind/M)
 	for(var/datum/role/R in members)
 		if(R.antag == M)
-			WARNING("Mind was already a role in this faction")
+			warning("Mind was already a role in this faction")
```
Comment (DamianX) [inline code/datums/gamemode/factions/faction.dm]: Why would you change these from `WARNING` (which shows file, line number, src, usr) to `warning` (which only shows the message)?

File: code/datums/gamemode/factions/faction.dm (hunk 4 above)
Comment (ShiftyRail) [reply code/datums/gamemode/factions/faction.dm]: Good point

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
Undo the changes to WARNING, remove the CRASHes, as they're supposed to get to the return 0s

**Comment by DamianX** [comment]
If you want to get runtimes without stopping the proc you can use `stack_trace()`

**Review by MadmanMartian — APPROVED** [review]

**Pull request merged**